Views/MenuItemLink-NotificationMenuItem.cshtml

```diff
@@ -0,0 +1,50 @@
+﻿@inject IOptions<AutorouteOptions> AutorouteOptions
+
+@using Microsoft.Extensions.Options;
+@using Microsoft.AspNetCore.Routing;
+@using OrchardCore.ContentManagement
+@using OrchardCore.ContentManagement.Routing;
+
+@{
+    ContentItem contentItem = Model.ContentItem;
+    var routeValues = new RouteValueDictionary(AutorouteOptions.Value.GlobalRouteValues);
+    routeValues[AutorouteOptions.Value.ContentItemIdKey] = Model.ContentItem.Content.NotificationMenuItem.SelectedContentItem.ContentItemIds[0];
+    var linkUrl = Url.RouteUrl(routeValues);
+    var currentPageUrl = FullRequestPath;
+
+    var query = (string)Model.ContentItem.Content.NotificationMenuItem.NotificationQuery.Value;
+    var notificationCount = 0;
+
+    try
+    {
+        notificationCount = (await Orchard.ContentQueryAsync(query)).Count();
+    }
+    catch
+    {
+        notificationCount = 0;
+    }
+
+    var cssClasses = "notification";

```

Comment:
I'm wondering if this is too generic/vague. Would it be clearer to have `notification-menu-item`? Or give all menu items a `menu-item` class and add `menu-item--with-notification` to this? Something like that?

Reply:
I think we would do that in `MenuItem.liquid`.

https://github.com/EtchUK/Etch.OrchardCore.Menu/blob/main/Views/MenuItem.liquid

Reply:
Updated the `MenuItem` template to add a `nav__item--content-type` modifier.